### PR TITLE
[subset] support option "--notdef-outline"

### PR DESCRIPTION
**Pull request by qxliu76**

### CODE REVIEW

File: src/hb-subset-input.hh [hunk 1]
```diff
@@ -45,12 +45,13 @@ struct hb_subset_input_t
   hb_set_t *drop_tables;
   hb_set_t *layout_features;
 
-  bool drop_hints;
-  bool desubroutinize;
-  bool retain_gids;
-  bool name_legacy;
-  bool overlaps_flag;
-  bool retain_all_layout_features;
+  int32_t drop_hints;
```
Comment (garretrieger) [inline src/hb-subset-input.hh:48]: Can you add a comment explaining why these are int32's intsead of booleans

**Review by garretrieger — APPROVED** [review]

File: src/hb-ot-glyf-table.hh [hunk 2]
```diff
@@ -209,7 +209,10 @@ struct glyf
 		if (!plan->old_gid_for_new_gid (new_gid, &subset_glyph.old_gid))
 		  return subset_glyph;
 
-		subset_glyph.source_glyph = glyf.glyph_for_gid (subset_glyph.old_gid, true);
+                if (new_gid == 0 && !plan->notdef_outline)
+                  subset_glyph.source_glyph = Glyph ();
+                else
```
Comment (behdad) [inline src/hb-ot-glyf-table.hh:214]: Tabs instead of spaces please.

File: src/hb-ot-var-gvar-table.hh [hunk 3]
```diff
@@ -461,6 +462,7 @@ struct gvar
       else
 	((HBUINT16 *) subset_offsets)[gid] = glyph_offset / 2;
 
+      if (gid == 0 && !c->plan->notdef_outline) continue;
```
Comment (behdad) [inline src/hb-ot-var-gvar-table.hh:465]: Can't you adjust the start of the loop instead?

**Comment by behdad** [comment]
LGTM w comments. Thanks.

**Comment by behdad** [comment]
Both loops in gvar can just adjust theirs start, instead of `gid = 0` can become `gid = 1` if notdef outline isn't desired. That's what I meant.

**Comment by qxliu76** [comment]
> Both loops in gvar can just adjust theirs start, instead of `gid = 0` can become `gid = 1` if notdef outline isn't desired. That's what I meant.

Done. :)

**Comment by behdad** [comment]
Lovely. Thanks.

**Pull request merged**